### ISSUE TITLE
HPA Targets an invalid object query is wrong

### ISSUE THREAD

**Issue by gravufo** [Checkmarx/kics#5990]
The KICS scan with Kubernetes type enabled displays an issue with my HPA when it is a fully valid HPA.

You can validate with the following file taken from the official [Kubernetes HPA walkthrough](https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/):
```yaml
apiVersion: autoscaling/v2
kind: HorizontalPodAutoscaler
metadata:
  name: php-apache
spec:
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: php-apache
  minReplicas: 1
  maxReplicas: 10
  metrics:
  - type: Resource
    resource:
      name: cpu
      target:
        type: Utilization
        averageUtilization: 50
```

I have checked the rego policy for this query and it specifically requires a `- type: Object` whereas `- type: Resource` is perfectly valid.
Here is the link to the rego file: https://github.com/Checkmarx/kics/blob/master/assets/queries/k8s/hpa_targets_invalid_object/query.rego

I think the test cases should include the example from the official walkthrough.

### Expected Behavior

No issue should be found on this file.

### Actual Behavior

```
HPA Targets Invalid Object
https://kubernetes.io/docs/tasks/run-application/horizontal-pod-autoscale-walkthrough/ 

The Horizontal Pod Autoscale must target a valid object

Platform: Kubernetes
Category: Availability
Query ID: 2f652c42-619d-4361-b361-9f599688f8ca

Results (1)
/helm/templates/hpa.yaml
Line: 13
Expected: spec.metrics[0] is a valid object
```

### Steps to Reproduce the Problem

Simply run KICS with the Kubernetes type enabled on a folder that contains the file pasted above.

### Specifications

-   Version: Latest
-   Platform: Linux

**Comment by gravufo** [comment]
Hey thanks for the quick resolution! I just tested it with the new version and it seems to work fine!

Have a great day :)
